### ISSUE TITLE
💅 false positive detection on lint/complexity/noUselessEscapeInRegex in 2.0.0-beta.6

### ISSUE THREAD

**Issue by flumm** [biomejs/biome#6201]
### Environment information

```bash
The following code produces a false positive:



const regex = /^[\\-]+$/;


it believes the '-' is escaped, while actually the '\' is escaped.
The safe fix, removes both backslashes, which results in wrong code
```

### Rule name

noUselessEscapeInRegex

### Playground link

https://next.biomejs.dev/playground/?indentStyle=space&indentWidth=4&lintRules=all&code=YwBvAG4AcwB0ACAAcgBlAGcAZQB4ACAAPQAgAC8AXgBbAFwAXAAtAF0AKwAkAC8AOwAKAAoAIgBcAFwAIgAuAG0AYQB0AGMAaAAoAHIAZQBnAGUAeAApADsA&language=js

### Expected result

it should not falsely detect the escape for '-'

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
